File: java/socketd-transport/socketd-transport-smartsocket/src/main/java/org/noear/socketd/transport/smartsocket/TcpAioFactory.java
package org.noear.socketd.transport.smartsocket;

import org.noear.socketd.transport.client.ClientFactory;
import org.noear.socketd.transport.server.ServerFactory;
import org.noear.socketd.transport.client.Client;
import org.noear.socketd.transport.client.ClientConfig;
import org.noear.socketd.transport.server.Server;
import org.noear.socketd.transport.server.ServerConfig;

/**
 * Tcp-Aio 经纪人实现
 *
 * @author noear
 * @since 2.0
 */
public class TcpAioFactory implements ClientFactory, ServerFactory {
    @Override
    public String[] schema() {
        return new String[]{"tcp", "tcps", "tcp-smartsocket"};
    }

    @Override
    public Server createServer(ServerConfig serverConfig) {
        return new TcpAioServer(serverConfig);
    }

    @Override
    public Client createClient(ClientConfig clientConfig) {
        return new TcpAioClient(clientConfig);
    }
}

File: python/socketd/socketd_aio_tcp/TcpAioFactory.py
from socketd.transport.client.ClientConfig import ClientConfig
from socketd.transport.server.ServerConfig import ServerConfig
from socketd.transport.client.Client import Client
from socketd.transport.client.ClientFactory import ClientFactory
from socketd.transport.server.Server import Server
from socketd.transport.server.ServerFactory import ServerFactory

from .TCPAIOServer import TCPAIOServer
from .TcpAioClient import TcpAioClient


class TcpAioFactory(ClientFactory, ServerFactory):

    def schema(self) -> list[str]:
        return ["tcp", "tcp-python", "std:tcp"]

    def create_server(self, serverConfig: ServerConfig) -> Server:
        return TCPAIOServer(serverConfig)

    def create_client(self, clientConfig: ClientConfig) -> Client:
        return TcpAioClient(clientConfig)
